node_modules/oae-principals/tests/test-emails.js

```diff
@@ -0,0 +1,603 @@
+            var tenantAlias = TenantsTestUtil.generateTestTenantAlias();
+            var tenantHost = tenantAlias + '.oae.com';
+            TestsUtil.createTenantWithAdmin(tenantAlias, tenantHost, function(err, tenant, tenantAdminRestContext) {
+                assert.ok(!err);
+
+                // Import users as a global admin using a local authentication strategy
+                PrincipalsTestUtil.importUsers(globalAdminRestContext, tenant.alias, getDataFileStream('users-emails.csv'), 'local', null, function(err) {
+                    assert.ok(!err);
+
+                    // Verify the user's email address is verified
+                    var restContext = TestsUtil.createTenantRestContext(tenant.host, 'users-emails-abc123', 'password');
+                    RestAPI.User.getMe(restContext, function(err, user) {
+                        assert.ok(!err);
+                        assert.strictEqual(user.email, 'foo@users.emails.com');
+                        assert.strictEqual(user.emailVerified, true);
+                        assert.ok(!user.emailChange);
+
+                        // Assert there's a mapping for the email address
+                        PrincipalsTestUtil.assertUserEmailMapping('foo@users.emails.com', [user.id], function() {
+
+                            // Update the email address through a CSV import
+                            PrincipalsTestUtil.importUsers(globalAdminRestContext, tenant.alias, getDataFileStream('users-emails-updated.csv'), 'local', true, function(err) {
+                                assert.ok(!err);
+
+                                RestAPI.User.getMe(restContext, function(err, user) {
+                                    assert.ok(!err);
+                                    assert.strictEqual(user.email, 'bar@users.emails.com');
+                                    assert.strictEqual(user.emailVerified, true);
+                                    assert.ok(!user.emailChange);
+
+                                    // Assert there's a mapping for the new email address
+                                    PrincipalsTestUtil.assertUserEmailMapping('bar@users.emails.com', [user.id], function() {
+
+                                        // Assert there's no mapping for the old email address
+                                        PrincipalsTestUtil.assertUserEmailMapping('foo@users.emails.com', [], function() {
+                                            return callback();
+                                        });
+                                    });
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+
+        /**
+         * Test that verifies that users updating their email address need to verify it
+         */
+        it('verify users updating their email address need to verify it', function(callback) {
```

Comment:
We should define and add tests for what happens when a user updates their email address, then updates to something different, then validate what happens when either verification is accepted.

My feeling is that a user should only have verifications pending for one email at a time, and once it is verified, all verifications are deleted. A user could have multiple verification codes for the same email (as is tested in this PR), but they can't have separate verification codes for separate email change requests. A new change deletes any old verifications.